Tools/autotest/quadplane.py

```diff
@@ -78,6 +78,144 @@ def get_disarm_delay(self):
     def set_autodisarm_delay(self, delay):
         self.set_parameter("LAND_DISARMDELAY", delay)
 
+    def get_servo_pwm(self, channel, min_pwm):
+        self.delay_sim_time(2)
+        self.wait_servo_channel_value(channel, min_pwm, comparator=operator.gt)
+        m = self.mav.recv_match(type='SERVO_OUTPUT_RAW', blocking=True, timeout=1)
+        if m is None:
+            raise NotAchievedException("Did not get SERVO_OUTPUT_RAW")
+        pwm = getattr(m, "servo5_raw", None)
+        return pwm
+
+    def test_airmode(self):
+        """Check that plane.air_mode turns on and off as required"""
+        self.progress("########## Testing AirMode operation")
+        self.set_parameter("AHRS_EKF_TYPE", 10)
+        self.change_mode('QSTABILIZE')
+        self.zero_throttle()
+        self.wait_ready_to_arm()
+
+        """SPIN_ARM and SPIN_MIN default to 0.10 and 0.15"""
+        """when armed with zero throttle in AirMode, motor PWM should be at SPIN_MIN"""
+        """If AirMode is off, motor PWM will drop to SPIN_ARM"""
+
+        """Make sure that SERVO5 is Motor1 (default)"""
+        assert(33 == self.get_parameter('SERVO5_FUNCTION'))
+
+        """Make sure that flightmode channel is 5 (default)"""
+        assert(5 == self.get_parameter("FLTMODE_CH"))
+
+        """When disarmed, motor PWM will drop to min_pwm"""
+        #min_pwm = self.get_parameter('SERVO5_MIN')
+        """SERVO5_MIN is not respected; it defaults to 1100, but disarmed PWM is 1000"""
+        min_pwm = 1000
+
+        """set Q_OPTIONS bit AIRMODE"""
+        self.set_parameter("Q_OPTIONS", 512)
+
+        """set  RC8_OPTION to ARMDISARM"""
+        self.set_parameter("RC8_OPTION", 41)
+
+        """arm with GCS, record Motor1 SPIN_ARM PWM output and disarm"""
+        self.arm_vehicle()
+        self.progress("Waiting for Motor1 to spool up to SPIN_ARM")
+        spin_arm_pwm = self.get_servo_pwm(5, min_pwm)
+        self.progress("spin_arm_pwm: %d" % spin_arm_pwm)
+        self.disarm_vehicle()
+
+        """arm with switch, record Motor1 SPIN_MIN PWM output and disarm"""
+        self.set_rc(8, 2000)
+        self.progress("Waiting for Motor1 to spool up to SPIN_MIN")
+        spin_min_pwm = self.get_servo_pwm(5, min_pwm)
+        self.progress("spin_min_pwm: %d" % spin_min_pwm)
+        self.set_rc(8, 1000)
+
+        assert(spin_arm_pwm < spin_min_pwm)
+
+        """verify that arming with switch results in higher PWM output"""
+        self.progress("Test auxswitch arming with Q_OPTIONS=AirMode")
+        modes = ('QSTABILIZE', 'QACRO')
```

Comment:
Maybe just inline this.

Reply:
wouldn't that add a dozen lines of code?

Reply:
Suggestion added.

Likewise for the loop below this one.